drone/jsonnet/jsonnet.go

```diff
@@ -97,31 +97,27 @@ func generate(c *cli.Context) error {
 			return err
 		}
 		for _, doc := range docs {
+			formatted, yErr := yaml.JSONToYAML([]byte(doc))
+			if yErr != nil {
+				return fmt.Errorf("failed to convert to YAML: %v", yErr)
+			}
 			buf.WriteString("---")
 			buf.WriteString("\n")
-			buf.WriteString(doc)
+			buf.WriteString(string(formatted))
 		}
 	} else {
 		result, err := vm.EvaluateSnippet(source, string(data))
 		if err != nil {
 			return err
 		}
-		buf.WriteString(result)
-	}
-
-	// enable yaml formatting with --format
-	if c.Bool("format") {
-		manifest, err := yaml.Parse(buf)
-		if err != nil {
-			return err
+		formatted, yErr := yaml.JSONToYAML([]byte(result))
+		if yErr != nil {
+			return fmt.Errorf("failed to convert to YAML: %v", yErr)
 		}
-		buf.Reset()
-		pretty.Print(buf, manifest)
+		buf.WriteString(string(formatted))
```

Comment:
you can use `buf.Write` instead of `buf.WriteString` to avoid the bytes to string conversion